R/step-first.R

```diff
@@ -61,6 +61,12 @@
 #'   filter(mpg < mean(mpg)) %>%
 #'   summarise(hp = mean(hp))
 lazy_dt <- function(x, name = NULL, immutable = TRUE, key_by = NULL) {
+  if (is.data.frame(x)) {
```

Comment:
You don't need this if statement since `group_vars()` already returns `character()` when applied to a data frame.

Reply:
ah, I should have added a comment there. I added this line because of a test in `test-step_call.R` where `lazy_dt()` is applied to a matrix. As there is no method `group_vars.matrix()` this errored. Do you prefer to add a method for matrices instead?

Reply:
If we added a method it would have to go into dplyr, which would be a pain.

Maybe just flip it and make the test `if (inherits(x, "grouped_df"))` ?

Reply:
Or `groups <- tryCatch(group_vars(x), error = function(e) character())`? 

I don't generally like `tryCatch()` for control flow, but I think it makes sense here since there's no requirement for `as.data.table()` and `group_vars()` to provide methods for the same classes.

Reply:
Makes sense to me. I also added a comment for clarification.